app/controllers/provider/admin/account/logos_controller.rb

```diff
@@ -1,38 +1,33 @@
 class Provider::Admin::Account::LogosController < Provider::Admin::Account::BaseController
   activate_menu! :audience, :cms, :logo
+  before_action :authorize, only: %i[destroy update]
 
   def edit
     @profile = profile
   end
 
   def update
-    authorize! :update, :logo
-
-    redirect_to edit_provider_admin_account_logo_path and return false if params[:profile].blank?
-
-    profile.update_attribute(:logo, params[:profile][:logo])
-    @notice = 'Logo was successfully created.'
-
-    respond_to do |format|
-      format.html do
-        flash[:notice] = @notice
-        redirect_to edit_provider_admin_account_logo_path
-      end
-
-      format.js
-    end
+    update_logo(logo: params.require(:profile).require(:logo))
   end
 
   def destroy
-    authorize! :delete, :logo
+    update_logo(logo: nil)
+  end
 
-    profile.update_attribute(:logo, nil)
+  private
 
-    flash[:notice] = 'Your logo was successfully deleted.'
+  def update_logo(logo:)
+    if profile.update({logo: logo})
+      flash[:notice] = t("#{action_name}_success", scope: 'logos_controller')
+    else
+      flash[:error] = profile.errors.full_messages.to_sentence
+    end
     redirect_to edit_provider_admin_account_logo_path
   end
 
-  private
+  def authorize
+    authorize! action_name, :logo
```

Comment:
Hmm for the `destroy` action it would be `authorize 'destroy', :logo` instead of `authorize :delete, :logo` as it was https://github.com/3scale/porta/pull/975/files#diff-5512c93ae956a903f6808306318f5dc3L27

Reply:
@hallelujah 
When I made this change I checked in the abilities to add the "destroy" there next to the "delete", and what I found is the only permission we are giving is actually "manage".

So I think we should not complicate the code knowing that.

I also added the tests for the conditions under it should be forbidden.